test/java/com/arpnetworking/metrics/portal/reports/impl/EmailSenderTest.java

```diff
@@ -100,17 +107,35 @@ public void testSendFailsIfExceptionThrown() throws MailException, ExecutionExce
         final RenderedReport report = TestBeanFactory.createRenderedReportBuilder().build();
         try {
             _sender.send(
-                    TestBeanFactory.createRecipient(),
+                    TestBeanFactory.createRecipientBuilder().build(),
                     ImmutableMap.of(new HtmlReportFormat.Builder().build(), report)
             ).toCompletableFuture().get();
         } catch (final ExecutionException e) {
-            if (e.getCause() instanceof MailException) {
-                throw (MailException) e.getCause();
+            final Throwable cause = e.getCause();
+            if (cause instanceof MailException) {
+                throw (MailException) cause;
             }
             throw e;    
         }
     }
 
+    @Test
+    public void testSendRejectsIllegalRecipients() {
+        final ImmutableList<String> legalAddresses = ImmutableList.of("alice@example.com", "bob@example.com");
+        for (final String address : legalAddresses) {
+            _sender.send(TestBeanFactory.createRecipientBuilder().setAddress(address).build(), ImmutableMap.of());
+        }
+
+        final ImmutableList<String> illegalAddresses = ImmutableList.of("", "@example.com", "charlie@example.com.ru");
```

Comment:
nit. Can you split this into its own test case? the top portion is more `testSendAllowsWhitelistedRecipients`

It _may_ also be useful to use JUnit rules here as well so that you can see all recipients that failed
https://junit.org/junit4/javadoc/latest/org/junit/rules/ErrorCollector.html